code/src/redux/reducers/bag.js

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+export const bag = createSlice({
+  name: 'bag',
+  initialState: {
+    items: [
+      //{ id: 1, name: 'Handduk', inBag: false },
+      //{ id: 2, name: 'Tandborste', inBag: false },
+      //{ id: 3, name: 'Tvål', inBag: false },
+    ]
+  },
+  reducers: {
+    addThings: (state, action) => {
+        const newItem = {  
+        id: Math.random(...state.items.map((item)=>item.id)),       
+        name: action.payload,
+        inBag: false
+        }
+        const newList = [...state.items, newItem]
+        state.items = newList;
+    },
+    removeItem: (state, action) => {
+      state.items = state.items.filter((item) => item.id !== action.payload)
+    },
+    tooglePacked: (state, action) => {
+      const foundThing = state.items.find((item) => item.id === action.payload)
+  
+      if (foundThing) {
+        foundThing.inBag = !foundThing.inBag
```

Comment:
As I do not understand this yet myself, but got some feedback from a "real" developer and maybe a teacher can help out here? Ok to mutate the state here like this, or should we create a new state first, and then change the state? And if yes, suggestions on how?

Reply:
You can read more about it [here](https://redux-toolkit.js.org/tutorials/intermediate-tutorial#mutable-update-logic). And more about immer that makes this possible [here](https://immerjs.github.io/immer/docs/introduction).